src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -32,12 +32,8 @@ const NavHeader = (props) => {
     initialData,
     opacityHeaderAnimation,
     onShareCommunity,
-    getTopicDetail,
-    setMemberCount,
-    memberCount,
     isHeaderHide,
     topicDetail,
-    setIsFollow,
     isFollow,
     hasSearch,
     searchText,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [32-51]

The removal of some props like `getTopicDetail`, `setMemberCount`, `memberCount`, `setIsFollow`, `getSearchLayout` and addition of new props like `onFollowButtonPress`, `followType` is a significant change. It seems like you're moving towards a more event-driven approach where parent components handle state changes, which can be a good practice for better separation of concerns. However, make sure that all components using this one have been updated accordingly to pass these new props and handle the removed ones if necessary.



<!-- This is an auto-generated comment by OSS CodeRabbit -->